src_c/polygon.c

```diff
@@ -468,6 +468,31 @@ pg_polygon_move(pgPolygonObject *self, PyObject *const *args, Py_ssize_t nargs)
     return tmp;
 }
 
+static PyObject *
+pg_polygon_as_segments(pgPolygonObject *self) {
+    PyObject *list = PyList_New(0);
+    for (Py_ssize_t i = 0; i < self->polygon.verts_num*2; i+=2) {
+        double next_vertex_x;
+        double next_vertex_y;
+
+        if (i >= self->polygon.verts_num * 2 - 2) {
+            next_vertex_x = self->polygon.vertices[0];
+            next_vertex_y = self->polygon.vertices[1];
+        }
+        else {
+            next_vertex_x = self->polygon.vertices[i + 2];
+            next_vertex_y = self->polygon.vertices[i + 3];
+        }
+
+        PyList_Append(list, pgLine_New4(
+            self->polygon.vertices[i], self->polygon.vertices[i + 1], 
+            next_vertex_x, next_vertex_y)
```

Comment:
This has two problems:
1. You should use `PyList_SET_ITEM`
2. You are not making sure that pgLine_New4() returns a valid object and not NULL